### DISCUSSION TITLE
API route crashes with 500 error

### DISCUSSION THREAD

**Discussion by LGmatrix13** [prisma/prisma#6418] [Q&A] [answered]
Hello everyone.

So, I am making an axios request alongside a prisma create function

Code: 
```ts
import axios from "axios";
import { getSession } from "next-auth/client";

import prisma from '../../lib/prisma'

const secretHandler = async (req, res) => {
  const session = await getSession({ req });
  if (session) {
    const res = await axios.get(`http://exampleapi.com`)
    console.log(res.avatar)
    const result = await prisma.post.create({
      data: {
        user: req.body.user,
        url: req.body.url,
        image: res.avatar
      },
    })
    res.json(result)
  } else {
    res.statusCode = 403;
    res.end(`Access denied`);
  }
};

export default secretHandler;
```

The API response from axios looks as follows :

```json
[
  {
    "id": "1",
    "createdAt": "2021-01-08T13:45:56.839Z",
    "name": "username",
    "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/vitor376/128.jpg"
  }
]
```
If I make a request to this route, it should get this Axios response and place res.avatar into my `data `object. It results, though, in a `500` error saying res.avatar is `undefined`. The console.log of `res.avatar` is also undefined. Could someone kindly guide me in the right direction?

**Comment by LGmatrix13** [comment, marked as answer]
Just had to identify the array in the response via `res.data[0].avatar`. This, however, conflicted with `res.json(result)`, so I changed it to `response.data[0].avatar`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
